### PR TITLE
Add parameter in addCheckbox to prepend or append the checkbox column

### PR DESCRIPTION
**Pull request by karmendra**
Add parameter in addCheckbox to prepend or append the checkbox column.
When I try to add a checkbox column as the first column as following it doesn't show the checkbox column at all, if I put the `addCheckbox()` after `columns` it comes as the last column.

      return $this->builder()
                  ->addCheckbox() 
                  ->columns($this->getColumns())

### CODE REVIEW

**Comment by karmendra** [comment]
Can someone review this please, CI has failed, its suggestions are not what we would expect in the code. Can someone confirm if we are waiting for CI to Pass before this PR is visible for review?

**Comment by Namoshek** [comment]
Just had a look at your PR and to me it looks fine (technically and style). I can confirm that the suggested StyleCI changes are not related to the changes of yours.

I just have one suggestion for an improvement: Maybe make the new parameter `@param bool|int` to also allow for positional inserts. Consider something like this:
```php
$this->builder()
     ->columns([
         ['data' => 'id', 'title' => 'ID'],
         ['data' => 'name', 'title' => 'Name'],
         ['data' => 'email', 'title' => 'Email'],
         ['data' => 'telephone', 'title' => 'Telephone'],
     ])
     ->addCheckbox(['data' => 'is_admin', 'title' => 'Admin?'], 2);
```
which would hopefully give you something like
```
 ID | Name | Admin? | Email            | Telephone
-----------------------------------------------------
  1 | Luke |    Yes | luke@example.com | 1234 567890
```
where `2` is the 0-based index of where the checkbox column should go.

Obviously this would require some edge case handling to cover when there is a too high index for example.

**Comment by yajra** [comment]
Sorry for overlooking this, a bit busy on mobile dev and cannot find time yet for open source projects. Anyways, will fix the StyleCI and will try to review this within the week. Thanks!

File: src/Html/Builder.php [hunk 1]
```diff
@@ -475,7 +476,12 @@ public function addCheckbox(array $attributes = [])
             'printable'      => true,
             'width'          => '10px',
         ], $attributes);
-        $this->collection->push(new Column($attributes));
+
+        if($prependColumn) {
```
Comment (yajra) [inline src/Html/Builder.php]: Please add a space after `if ($prependColumn)`

**Comment by karmendra** [comment]
@Namoshek Very good suggestion, but I am not sure how I can insert an item in Collection at specific index. I quick search showed that we can use splice prepend/push and union functions to do so, but felt that's really not good way to do it. Any other suggestion do let me know.

File: src/Html/Builder.php (hunk 1 above)
Comment (karmendra) [reply src/Html/Builder.php]: Added a space after if

**Comment by Namoshek** [comment]
Yes, `$collection->splice($index, $length, $elementsToInsert)` is the way to go, I think. Although you could also build a new collection up from ground if that's easier for you (due to the very limited amount of entries for the collection, this would not be a performance issue either).

I think something along these lines could be a potential solution (completely untested of though):
```php
/**
 * Add a checkbox column. Prepends the column at the beginning,
 * appends it at the end or inserts it at a specified index.
 *
 * @param  array $attributes
 * @param  bool|int $position true to prepend, false to append or a zero-based index for positioning
 * @return $this
 */
public function addCheckbox(array $attributes = [], $position = false)
{
    $column = new Column(array_merge([
        'defaultContent' => '<input type="checkbox" ' . $this->html->attributes($attributes) . '/>',
        'title'          => $this->form->checkbox('', '', false, ['id' => 'dataTablesCheckbox']),
        'data'           => 'checkbox',
        'name'           => 'checkbox',
        'orderable'      => false,
        'searchable'     => false,
        'exportable'     => false,
        'printable'      => true,
        'width'          => '10px',
    ], $attributes));

    if ($position === true) {
        $this->collection->prepend($column);
    } else if ($position === false || $position >= $this->collection->count()) {
        $this->collection->push($column);
    } else {
        $this->collection->splice($position, 0, [$column]);
    }
    
    return $this;
}
```

If you are curious what the equivalent of the `if-elseif-else` part is:
```php
if (is_int($position)) {
    if ($position >= $this->collection->count()) {
        $this->collection->push($column);
    } else {
        $this->collection->splice($position, 0, [$column]);
    }
} else {
    if ($position) {
        $this->collection->prepend($column);
    } else {
        $this->collection->push($column);
    }
}
```
But I find the above version more readeable and also a bit more logic when thinking about it twice. It makes use of implicit type checks though (`===` vs. `is_int()`).

**Comment by karmendra** [comment]
@Namoshek I implemented your suggestion and tested it as well. worked like a charm.
@yajra please review and let us know your suggestions if any.

File: src/Html/Builder.php [hunk 2]
```diff
@@ -418,7 +419,14 @@ public function addCheckbox(array $attributes = [])
             'printable'      => true,
             'width'          => '10px',
         ], $attributes);
-        $this->collection->push(new Column($attributes));
+        $column = new Column($attributes);
+	    if ($position === true) {
```
Comment (Namoshek) [inline src/Html/Builder.php]: I think you mixed up the indent from line 423 to 429 a little bit, there is a tabstop involved and the indent is off by 1.

File: src/Html/Builder.php [hunk 3]
```diff
@@ -418,7 +419,14 @@ public function addCheckbox(array $attributes = [])
             'printable'      => true,
             'width'          => '10px',
         ], $attributes);
-        $this->collection->push(new Column($attributes));
+        $column = new Column($attributes);
```
Comment (Namoshek) [inline src/Html/Builder.php]: Maybe add an empty line after `$column = new Column($attributes);`, StyleCI most likely doesn't like it like this. 👍

**Comment by karmendra** [comment]
@yajra can you please check why it is still showing "1 review requesting changes" for one of your review. I already made the required changes.

**Comment by yajra** [comment]
Sorry for the delays, released on v3.7.1 thanks!

**Pull request merged**